napari/utils/misc.py

```diff
@@ -456,10 +458,15 @@ def ensure_layer_data_tuple(val):
     return val
 
 
-def ensure_list_of_layer_data_tuple(val) -> List[tuple]:
+def ensure_list_of_named_layer_data_tuple(val) -> List[LayerDataTupleNamed]:
+    """
+    Make sure *val* is a list of LayerDataTupleNamed objects.
+    """
     if isinstance(val, list) and len(val):
         with contextlib.suppress(TypeError):
-            return [ensure_layer_data_tuple(v) for v in val]
+            return [
+                LayerDataTupleNamed(ensure_layer_data_tuple(v)) for v in val
+            ]
     raise TypeError(
         trans._('Not a valid list of layer data tuples!', deferred=True)
```

Comment:
```suggestion
        trans._('Not a valid list of layer data tuples !  ({val})',
            val=val,
            deferred=True
        )
```

This will help to debug when this fails.